cpp/src/umap/fuzzy_simpl_set/naive.cuh

```diff
@@ -328,7 +331,8 @@ void launcher(int n,
    * Compute graph of membership strengths
    */
 
-  dim3 grid_elm(raft::ceildiv(n * n_neighbors, TPB_X), 1, 1);
+  uint64_t to_process = static_cast<uint64_t>(in.n_rows) * n_neighbors;
+  dim3 grid_elm(raft::ceildiv(to_process, TPB_X), 1, 1);
```

Comment:
I believer there's a chance for an overflow here. If `to_process` is large enough, we'd get something that would overflow a 32-bit integer, right? We also probably want to limit this to the maximum number of blocks (65535) anyway.

Reply:
Regarding integer overflow, if at least one number in the multiplication is on 64 bits, the other is automatically casted by the compiler. Unless, I am missing something it should be fine in this case. Regarding the maximal grid dimension, I believe that on GPUs with Compute Capability 2.x and higher `gridDim.x` should be able to go up to 2³¹ – 1.

Reply:
@viclafargue TPB is threads per block. There's a limited number of threads in a physal warp and a limited number of warps that can be logically grouped into a block. The number of warps that can be run on an SM concurrently is usually 64, and the number that can be grouped into a block is usually 32 (max 1024 threads per block). This number does not need to be 64-bit. Please cast in place here. It's in host memory. You don't need to promote the template type for this.